src/hb-ot-name-table.hh

```diff
@@ -97,12 +97,30 @@ struct NameRecord
     return UNSUPPORTED;
   }
 
-  NameRecord* copy (hb_serialize_context_t *c, const void *base) const
+  NameRecord* copy (hb_serialize_context_t *c, const void *base,
+                    const hb_hashmap_t<unsigned, const char*> *name_table_overrides) const
   {
     TRACE_SERIALIZE (this);
+    auto snap = c->snapshot ();
     auto *out = c->embed (this);
     if (unlikely (!out)) return_trace (nullptr);
-    out->offset.serialize_copy (c, offset, base, 0, hb_serialize_context_t::Tail, length);
+    if (name_table_overrides->has (nameID)) {
+      const char *name_str = name_table_overrides->get (nameID);
+      unsigned len = strlen (name_str);
+      if (!c->check_assign (out->length, len, HB_SERIALIZE_ERROR_INT_OVERFLOW)) {
+        c->revert (snap);
+        return_trace (nullptr);
+      }
+
+      hb_bytes_t name_bytes (name_str, len);
+
+      out->offset = 0;
+      c->push ();
+      name_bytes.copy (c);
```

Comment:
You'll need to convert the strings into UTF-16BE before storage, the spec says the strings are:

"Strings for the Unicode platform must be encoded in UTF-16BE."

@behdad is there an existing method in harfbuzz to encode the string into UTF-16BE?

Reply:
Can we assume the input string is encoded in UTF-16BE already? We can do this conversion in JAVA

Reply:
I don't think a harfbuzz API should require that. It should take UTF-16 (platform-endian) or UTF-8.

Reply:
It encodes one character at a time.

Reply:
My preference is to have the API take UTF-8 and then do encoding to UTF-16BE as that is most likely what callers will be using (also fine with having a second version of the method that can take a UTF-16)

Reply:
Don't expose it. Just use it internally.

Reply:
Done. thanks!